java/src/main/native/src/row_conversion.cu

```diff
@@ -187,6 +189,63 @@ struct batch_data {
   std::vector<row_batch> row_batches; // information about each batch such as byte count
 };
 
+rmm::device_uvector<size_type> build_string_row_sizes(table_view const &tbl,
+                                                      size_type fixed_width_and_validity_size,
+                                                      rmm::cuda_stream_view stream) {
+  auto const num_rows = tbl.num_rows();
+  rmm::device_uvector<size_type> d_row_sizes(num_rows, stream);
+
+  std::vector<strings_column_view::offset_iterator> offsets_columns;
```

Comment:
Nit:  `offset_iterators` might be more clear here, although maybe too confusing when used with `offsets_iter` below? Hmm.

Reply:
> maybe too confusing when used with `offsets_iter` below? Hmm.

It needn't be, actually. `offset_columns`, `offsets_iter`, etc. all exist in service of setting up `d_offsets_column`. Were they all to be moved into an IIFE, we might avoid that confusion.